homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,216 @@
+        match value:
+            case SystemMode.kAuto:
+                return HVACMode.AUTO
+            case SystemMode.kDry:
+                return HVACMode.DRY
+            case SystemMode.kFanOnly:
+                return HVACMode.FAN_ONLY
+            case SystemMode.kCool | SystemMode.kPrecooling:
+                return HVACMode.COOL
+            case SystemMode.kHeat | SystemMode.kEmergencyHeat:
+                return HVACMode.HEAT
+            case _:
+                return HVACMode.OFF
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Return the list of (currently) available hvac operation modes."""
+        value = int(
+            self.get_matter_attribute_value(
+                clusters.Thermostat.Attributes.ControlSequenceOfOperation
+            )
+        )
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingAndHeating,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingAndHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT, HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingWithReheat,
+        ):
+            return [HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT]
+        return [HVACMode.OFF]
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current running hvac operation (if supported)."""
+        if running_state := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningState
+        ):
+            if running_state == 0:
+                return HVACAction.HEATING
+            if running_state == 1:
+                return HVACAction.COOLING
+            if running_state == 2:
+                return HVACAction.FAN
+        if running_mode := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningMode
+        ):
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kCool:
+                return HVACAction.COOLING
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kHeat:
+                return HVACAction.HEATING
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kOff:
+                return HVACAction.OFF
+        return None
+
+    def _get_temperature_in_degrees(self, attribute) -> float | None:
+        """Return the scaled temperature value for the given attribute."""
+        if value := self.get_matter_attribute_value(attribute):
+            return value / TEMPERATURE_SCALING_FACTOR
+        return None
+
+    @property
+    def current_temperature(self) -> float | None:
```

Comment:
Can we update current temperature and target temperature in `_update_from_device` instead? Ie use `_attr_current_temperature` etc.

Reply:
I've put the logic in here instead of the "_update_from_device" to prevent that method getting too crowded as these are all one-off attributes that are not read multiple times. My personal method is to do calculations that are needed multiple times in the "_update_from_device" and one-off stuff, in the class properties for readability

Reply:
I think it's more readable to update all the attributes in the same place in the same event handler.

Reply:
I must say I personally don't agree as the result will/could be a very crowded callback with lots of statements in there because these properties need some sort of calculation. But that could be a personal preference then.

Maybe we should split them out again in separate subhandlers again if we reach X number of statements within the callback, and then I don't see the benefit over doing it within the property itself again :-)

Anyways, to avoid further discussion we could add them to the callback now but we must be aware of that callback not getting unreadable due to too many statements as my experience is that the climate platform has the habit of expanding rapidly with all exceptions and possibilities.

Reply:
@hidaris want me to move the attributes ?
also, the server bump is merged so if you rebase, you get the new write_attribute in the client